packages/upgrade/src/jscodeshift.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright IBM Corp. 2019, 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import which from 'npm-which';
+import spawn from './spawn';
+import path from 'path';
+
+let jscodeshift;
+
+function run(transformFileName, options) {
+  console.log(options);
+  if (!jscodeshift) {
+    jscodeshift = which.sync('jscodeshift', {
+      cwd: __dirname,
+    });
+  }
```

Comment:
My vote for finding the bin path would be to not change the workspace config for hoisting limits and find the bin path directly. Maybe something like:

```js
// tools/jscodeshift.js
import path from 'path';
import spawn from './spawn';

let _jscodeshift;

function getBinPath() {
  if (!_jscodeshift) {
    const directory = path.dirname(require.resolve('jscodeshift'));
    _jscodeshift = path.join(directory, 'bin', 'jscodeshift.js');
  }
  return _jscodeshift;
}

export async function run(options) {
  const { cwd, stdio = 'inherit', parser = 'babel', paths, transform } = options;
  const args = [
    paths,
    `-t=${transform}`,
    `--parser=${parser}`, 
    `--ignore-pattern=**/build/**`,
    `--ignore-pattern=**/dist/**`,
    `--ignore-pattern=**/es/**`,
    `--ignore-pattern=**/lib/**`,
    `--ignore-pattern=**/node_modules/**`,
    `--ignore-pattern=**/storybook-static/**`,
    `--ignore-pattern=**/umd/**`,
  ];

  if (options.print) {
    args.push('--print');
  }

  // Not sure what log level this would map to
  if (options.verbose) {
    args.push('-v');
  }

  if (options.dry) {
    args.push('--dry');
  }

  return await spawn(getBinPath(), args, {
    cwd,
    stdio,
  });
}

// Usage
import jscodeshift from './tools/jscodeshift';

async function command() {
  await jscodeshift.run({
    // options...
  });
}
```